packages/terra-form/src/Select.jsx

```diff
@@ -7,9 +7,18 @@ import './Select.scss';
 
 const propTypes = {
   /**
-   * List of choices to be selected.
+   * List of object key/value pairs for choices to be selected.
    */
-  choices: PropTypes.array.isRequired,
+  choices: PropTypes.arrayOf(PropTypes.shape({
+    value: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+    ]).isRequired,
+    display: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
```

Comment:
Can we lock down the display type to just strings? Or do numbers make things easier in some cases?